plugins/out_opensearch/opensearch.c

```diff
@@ -347,21 +388,36 @@ static int opensearch_format(struct flb_config *config,
      * this payload.
      */
 
-    enabled_data_stream_no_ra = ctx->data_stream_mode == FLB_TRUE && ctx->ra_data_stream == NULL;
-    disabled_logstash_fmt_no_ra = ctx->logstash_format == FLB_FALSE && ctx->ra_index == NULL;
+    if (ctx->data_stream_mode == FLB_TRUE &&
+        ctx->ra_data_stream == NULL) {
+        enabled_data_stream_no_ra = FLB_TRUE;
+    }
+    else {
+        enabled_data_stream_no_ra = FLB_FALSE;
+    }
+
+    if (ctx->logstash_format == FLB_FALSE &&
+        ctx->ra_index == NULL) {
+        disabled_logstash_fmt_no_ra = FLB_TRUE;
+    }
+    else {
+        disabled_logstash_fmt_no_ra =  FLB_FALSE;
+    }
 
-    if ((enabled_data_stream_no_ra || disabled_logstash_fmt_no_ra) && ctx->generate_id == FLB_FALSE) {
+    if ((enabled_data_stream_no_ra ==  FLB_TRUE ||
+    disabled_logstash_fmt_no_ra == FLB_TRUE) &&
```

Comment:
indent this line and the next one appropriately so they are visually contextually consistent.

Reply:
Fixed